### PR TITLE
use a11y_aria_label/a11y_wrap_focus helpers where applicable

### PR DESCRIPTION
**Pull request by chris-steele**
Addresses #1513

### CODE REVIEW

**Comment by chris-steele** [comment]
aria label helpers are causing issues with accessibility with JAWS

**Comment by chris-steele** [comment]
Fixed focus issue when using helper by removing `role="main"` from `page.hbs`. This attribute would have been removed by `$.fn.a11y_aria_label` anyway.

**Comment by oliverfoster** [comment]
@chris-steele 
can you also look at the wrap_focus / focus_guard stuff? a11y_wrap_focus helper + https://github.com/adaptlearning/adapt_framework/blob/master/src/core/js/libraries/jquery.a11y.js#L626-L659

basically, every popup should define it's own:
```{{a11y_wrap_focus}}```

as an interim the jquery.a11y should look for the existence of the focusguard element and inject it if it doesn't exist.

current popups are:
drawer
notify
hotgraphic

File: src/core/templates/notifyPush.hbs [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {{! make the _globals object in course.json available to this template}}
 {{import_globals}}
-<div class="notify-push-inner" role="region" aria-label="{{_globals._accessibility._ariaLabels.feedbackPopUp}}">
-	
+<div class="notify-push-inner" role="region">
```
Comment (oliverfoster) [inline src/core/templates/notifyPush.hbs]: the role=region can go

File: src/core/templates/notify.hbs [hunk 2]
```diff
@@ -1,6 +1,7 @@
 {{! make the _globals object in course.json available to this template}}
 {{import_globals}}
-<div {{#each _attributes}}{{@key}}="{{this}}" {{/each}} class="notify-popup notify-type-{{_type}} {{_classes}}" role="region" aria-label="{{_globals._accessibility._ariaLabels.feedbackPopUp}}">
+<div {{#each _attributes}}{{@key}}="{{this}}" {{/each}} class="notify-popup notify-type-{{_type}} {{_classes}}" role="region">
```
Comment (oliverfoster) [inline src/core/templates/notify.hbs]: role=region can be removed

File: src/core/templates/navigation.hbs [hunk 3]
```diff
@@ -1,7 +1,8 @@
 {{! make the _globals object in course.json available to this template}}
 {{import_globals}}
 {{! All links in the nav bar should have an attribute of data-event, navigationView uses this to fire a global event}}
-<div class="navigation-inner clearfix" role="navigation" aria-label="{{_globals._accessibility._ariaLabels.navigation}}">
+<div class="navigation-inner clearfix" role="navigation">
```
Comment (oliverfoster) [inline src/core/templates/navigation.hbs]: role=navigation needs to go

File: src/core/templates/notifyPush.hbs (hunk 1 above)
Comment (chris-steele) [reply src/core/templates/notifyPush.hbs]: Done

File: src/core/templates/notify.hbs (hunk 2 above)
Comment (chris-steele) [reply src/core/templates/notify.hbs]: Done

File: src/core/templates/navigation.hbs (hunk 3 above)
Comment (chris-steele) [reply src/core/templates/navigation.hbs]: Done

**Comment by chris-steele** [comment]
Popups now use helper for `focusguard`. As part of this work I've have done the following few tweaks:
- `focusguard` in each layer and now uses delegated events
- `isScrollDisabledOnPopupEnabled`: properly handle cases where there is more than one popup layer (e.g. overlays that contain questions which in turn display feedback)
- `a11y_popdown` will respect any `popup:closed` listener that moves focus
- prevent superfluous `popup:closed` events removing `body` from `floorStack`

File: src/core/js/libraries/jquery.a11y.js [hunk 4]
```diff
@@ -681,6 +669,34 @@
...
         }
 
+        function a11y_setupFocusGuard() {
+            var options = $.a11y.options;
+
+            if ($.a11y.state.isFocusGuardSetup) return;
+
+            $.a11y.state.isFocusGuardSetup = true;
+
+            $('body').on("click", domSelectors.focusguard, function(event) {
```
Comment (oliverfoster) [inline src/core/js/libraries/jquery.a11y.js]: can you pull these two together (click + focus) into:
```js
            $('body').on("click focus", domSelectors.focusguard, function(event) {

                if (options.isDebug) console.log ("focusguard");

                preventDefault(event);
                $.a11y_focus(true);

                return false;

            });
```

File: src/core/templates/navigation.hbs [hunk 5]
```diff
@@ -1,7 +1,8 @@
 {{! make the _globals object in course.json available to this template}}
 {{import_globals}}
 {{! All links in the nav bar should have an attribute of data-event, navigationView uses this to fire a global event}}
-<div class="navigation-inner clearfix" role="navigation" aria-label="{{_globals._accessibility._ariaLabels.navigation}}">
+<div class="navigation-inner clearfix">
+	{{{a11y_aria_label _globals._accessibility._ariaLabels.navigation}}}
```
Comment (oliverfoster) [inline src/core/templates/navigation.hbs]: can you normalise the indentation?

**Comment by oliverfoster** [comment]
@chris-steele is totally messed this up by idiotically pushing when i shouldn't have, i've made a new pr at https://github.com/adaptlearning/adapt_framework/pull/1940

**Pull request closed without merging**